pkg/tracegen/intake.go

```diff
@@ -21,28 +21,29 @@ package tracegen
 
 import (
 	"context"
+	"crypto/tls"
 	"errors"
 	"fmt"
+	"net/url"
 	"time"
 
 	"go.elastic.co/apm/v2"
+	"go.elastic.co/apm/v2/transport"
 )
 
 // SendIntakeV2Trace generate a trace including a transaction, a span and an error
 func SendIntakeV2Trace(ctx context.Context, cfg Config) (apm.TraceContext, error) {
-	var errs []error
 	if err := cfg.validate(); err != nil {
-		errs = append(errs, err)
+		return apm.TraceContext{}, err
 	}
-	if cfg.elasticAPMTracer == nil {
-		errs = append(
-			errs,
-			errors.New("elasticAPMTracer must be provided"),
-		)
-	}
-	if len(errs) > 0 {
-		return apm.TraceContext{}, errors.Join(errs...)
+
+	tracer, err := newTracer(cfg)
+	if err != nil {
+		return apm.TraceContext{}, fmt.Errorf("failed to create tracer: %w", err)
 	}
+	defer tracer.Close()
+	defer tracer.Flush(ctx.Done())
```

Comment:
This'll result in not being able to call `tracer.Stats()` in case we want to assert how many transactions, spans, errors, etc, We're doing this today in some tests.

To address this we can either:

- Return the `tracer.Stats`
- Return something that looks like tracer starts and tells you what was exported. (This'll also allow us to check what was exported for OTel.